spotlight-client/src/charts/BarChartCluster.tsx

```diff
@@ -0,0 +1,214 @@
+// Recidiviz - a data platform for criminal justice reform
+// Copyright (C) 2022 Recidiviz, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+import React, { useState, useCallback } from "react";
+import OrdinalFrame from "semiotic/lib/OrdinalFrame";
+import useBreakpoint from "@w11r/use-breakpoint";
+import { animated, useTransition } from "react-spring/web.cjs";
+import styled, { css } from "styled-components/macro";
+import { ChartWrapper as BaseChartWrapper } from "@recidiviz/design-system";
+import ColorLegend from "./ColorLegend";
+import ResponsiveTooltipController, {
+  ResponsiveTooltipControllerProps,
+} from "./ResponsiveTooltipController";
+import { highlightFade, useHighlightedItem } from "./utils";
+import MeasureWidth from "../MeasureWidth";
+import { animation, colors, zIndex } from "../UiLibrary";
+import { formatAsPct } from "../utils";
+import { RateByCategoryAndDemographicsRecords } from "../contentModels/types";
+
+const CHART_HEIGHT = 400;
+const CHART_MIN_WIDTH = 736;
+
+const Container = styled.figure`
+  width: 100%;
+`;
+
+const horizontalScrollStyles = css`
+  overflow-x: overlay !important;
+  overflow-x: auto;
+  overflow-y: hidden;
+`;
+
+const ChartWrapper = styled(BaseChartWrapper)<{
+  isScrollable?: boolean;
+}>`
+  position: relative;
+  z-index: ${zIndex.base + 1};
+
+  .foreground-graphics .ordinal-labels text {
+    transform: rotate(-45deg) translate(0rem, 0rem);
+  }
+
+  ${(props) => props.isScrollable && horizontalScrollStyles}
+
+  .BarChartCluster__segment {
+    stroke: ${colors.background};
+    stroke-width: 2;
+  }
```

Comment:
is there a reason you can't just set these properties as part of the `style` prop instead of adding the additional class? it's better not to do both on the same element if we can help it, just for easier readability

Reply:
The style prop is already contains complex logic of highlighting, and if use this styles there, we need to copy same lines several times.

Reply:
OK fair enough!